### ISSUE TITLE
Labeling in HTML Reports View Documentation

### ISSUE THREAD

**Issue by New-arch** [archimatetool/archi#864]
Feature Requirement: 
As HTML report provides us with HTML / Markdown Render , it provides a great way to compliment the viewpoint with content. 
However the HTML render from the view Documentation tag does not include any metadata from the model. Therefore there is no reuse of the model artefacts or the view properties within the html content. 

Feature requirement is to use view's properties in the html setup similar to what we use in labels on the objects presentation. 
A stretch target will be to use an object's properties in the view's render. 

<Table>
<tr VALIGN=Middle>
<td>
<h2>Architecture Decision Record Card</h2>
</td>
<td>
<h4> ID :${property:ADR_ID}  </h4>
</td>
<td>
CRITICALITY : <h4> ${property:viewProperty}   </h4>
</td>
<td>
DATE : <h4> 2022/08/20  </h4>
</td>
<td>
STATUS : <h4> Green  </h4>
</td>
</tr>

**Comment by Phillipus** [comment]
All of the code for Label Expressions is designed only to render in the objects' text in a View, or nodes in the Models Tree. I think trying to implement this in the Documentation field so that the expressions rendered in an HTML report would require some serious effort. Perhaps something to think about one day.

**Comment by New-arch** [comment]
Yes . Added a feature requirement to consider for the backlog.  We are using the html render more and more to augment the diagrams with content , rather than just diagrams and go to other design tools for content. This feature will improve the usability of the html report  as a more hoisting solution offering where we can bring in lists and matrixes to the repository managed by arch file with reusing the models artefacts in properties possibly. This extends the view's usage to explain something to a specific user community .

**Comment by Phillipus** [comment]
It turns out that an initial implementation of rendering label expressions in documentation fields in HTML and Jasper Reports was not so complicated:

https://github.com/archimatetool/archi/commit/f987970fbd0a4e9f221e053bad672f7a96665fa1

However:

- Is this something that other people want?
- Are there side-effects?
- What is the impact of this change?

We need to think about this.

**Comment by New-arch** [comment]
I use the html report allot and content to compliment the solution architecture viewpoints instead of running with separate pptx  for stakeholders that need more information or different content mediums.  Other architects recommended we move to C4Builder that gives you the combination of markdown files and plantuml for diagraming within one solution. I would rather stay with Archi.

**Comment by bratgers** [comment]
> It turns out that an initial implementation of rendering label expressions in documentation fields in HTML and Jasper Reports was not so complicated:
> 
> [f987970](https://github.com/archimatetool/archi/commit/f987970fbd0a4e9f221e053bad672f7a96665fa1)
> 
> However:
> 
>     * Is this something that other people want?
> 
>     * Are there side-effects?
> 
>     * What is the impact of this change?
> 
> 
> We need feedback on this.

For my new assignment I used Archi again, after several years. I like the simplicity and the power. of the tool I think it could be a great advantage if we could harmonize and unify the documentation and properties parts with a Markdown render, for all the output and report forms. 

For my previous assignment it was necessary to use BizzDesign and in my  opinion BizzDesign has three advances over Archi:

1. The ability to format text.
2. The ability to use references (hyperlinks) to elements in the documentation field. 
3. The ability to use vertical oriented  text labels.

I use the HTML and Jasper reporting tools and jArchi scripts to produce out in the same look and feel as the documentation of the customer. Te lack of formatting text is annoying.  As a first step, I think it it is a big plus if the documentation fields support Markdown. As a second step maybe it is possible to support internal linking to elements and maybe in the future vertical orientation for text labels.

There is probably one issue, Jasper Report support Markup formatting, and own implementation of of text fromating, beside html and rtf.  For the use of Markdown in Jasper Report the Markdown needs to be converted to html , rtf or the native Markkup format.
See: https://jasperreports.sourceforge.net/sample.reference/markup/index.html#markup

**Comment by evlibra** [comment]
> > It turns out that an initial implementation of rendering label expressions in documentation fields in HTML and Jasper Reports was not so complicated:
> > [f987970](https://github.com/archimatetool/archi/commit/f987970fbd0a4e9f221e053bad672f7a96665fa1)
> > However:
> > ```
> > * Is this something that other people want?
> > 
> > * Are there side-effects?
> > 
> > * What is the impact of this change?
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > We need feedback on this.
> 
> For my new assignment I used Archi again, after several years. I like the simplicity and the power. of the tool I think it could be a great advantage if we could harmonize and unify the documentation and properties parts with a Markdown render, for all the output and report forms.
> 
> For my previous assignment it was necessary to use BizzDesign and in my opinion BizzDesign has three advances over Archi:
> 
> 1. The ability to format text.
> 2. The ability to use references (hyperlinks) to elements in the documentation field.
> 3. The ability to use vertical oriented  text labels.
> 
> I use the HTML and Jasper reporting tools and jArchi scripts to produce output with the same look and feel as the documentation of the customer. The lack of formatting text is annoying. As a first step, I think it it is a big plus if the documentation fields support Markdown.
> 
> As a second step maybe it is possible to support internal linking to elements and maybe in the future vertical orientation for text labels.
> 
> There is probably one issue, Jasper Report support Markup formatting (the option styles). Markup is an own implementation of text formatting, beside html and rtf. For the use of Markdown in Jasper Report the Markdown needs to be converted to html , rtf or the native Markkup format. See: https://jasperreports.sourceforge.net/sample.reference/markup/index.html#markup

What about usage of reStructuredText for the report? See https://docutils.sourceforge.io/rst.html
Advantage would be ability to edit pages independently, even directly on the web, e.g. at GitHub. For that also would be good to include element name in the page name at the beginning of name (in addition to the id and element type, or instead of the element type)
